cmd/check_senders.cpp

```diff
@@ -31,24 +28,14 @@
 #include <silkworm/concurrency/worker.hpp>
 #include <silkworm/crypto/ecdsa.hpp>
 #include <silkworm/db/access_layer.hpp>
-#include <silkworm/db/mdbx.hpp>
 #include <silkworm/db/stages.hpp>
-#include <silkworm/db/util.hpp>
 #include <silkworm/etl/collector.hpp>
 #include <silkworm/stagedsync/recovery/recovery_farm.hpp>
-#include <silkworm/types/block.hpp>
 
 using namespace silkworm;
 
 std::unique_ptr<stagedsync::recovery::RecoveryFarm> farm;
 
-void sig_handler(int) {
-    std::cout << std::endl << " Got interrupt. Stopping ..." << std::endl << std::endl;
-    if (farm) {
-        farm->stop();
```

Comment:
Is the call to `farm->stop();` not needed anymore?

Reply:
Nice change overall!

Reply:
Hmm. It was already removed in commit 2b0cd28
Maybe you were reviewing the initial commit ?

Reply:
yes, it was removed. But before when handling a signal we called `farm->stop();`, and we don't call it anymore in the new version. Just wondering if it is not needed anymore?

Reply:
Oh I see your point. You refer to method `stop()` for the farm class.
Well this is still needed : should any worker return an error in the recovery phase then the whole farm is stopped. This is a programmatic stop, not a signalled one.
See https://github.com/torquem-ch/silkworm/blob/master/node/silkworm/stagedsync/recovery/recovery_farm.cpp#L231-L303